src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -192,56 +238,60 @@ void func_80989204(EnDg* this, PlayState* play) {
+    f32 ny;
+    f32 nz;
 
     if (this->actor.floorPoly != NULL) {
-        sp24 = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.x);
-        sp20 = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.y);
-        sp1C = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.z);
+        nx = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.x);
+        ny = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.y);
+        nz = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.z);
 
         sinf(0.0f);
         cosf(0.0f);
-        arg1->x = -Math_Acot2F(1.0f, -sp1C * sp20);
-        arg1->z = Math_Acot2F(1.0f, -sp24 * sp20);
+        floorRot->x = -Math_Acot2F(1.0f, -nz * ny);
+        floorRot->z = Math_Acot2F(1.0f, -nx * ny);
     }
 }
 
-s32 func_80989418(EnDg* this, Path* arg1, s32 arg2) {
-    Vec3s* sp5C = Lib_SegmentedToVirtual(arg1->points);
-    s32 count = arg1->count;
-    s32 idx = arg2;
-    s32 sp50 = false;
-    f32 phi_f12;
-    f32 phi_f14;
-    f32 sp44;
-    f32 sp40;
-    f32 sp3C;
-    Vec3f sp30;
-
-    Math_Vec3s_ToVec3f(&sp30, &sp5C[idx]);
-    if (idx == 0) {
-        phi_f12 = sp5C[1].x - sp5C[0].x;
-        phi_f14 = sp5C[1].z - sp5C[0].z;
-    } else if (idx == count - 1) {
-        phi_f12 = sp5C[count - 1].x - sp5C[count - 2].x;
-        phi_f14 = sp5C[count - 1].z - sp5C[count - 2].z;
+s32 EnDg_HasReachedPoint(EnDg* this, Path* path, s32 pointIndex) {
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    s32 pathCount = path->count;
+    s32 currentPoint = pointIndex;
+    s32 reached = false;
+    f32 diffX;
+    f32 diffZ;
+    f32 px;
+    f32 pz;
+    f32 d;
+    Vec3f point;
+
+    Math_Vec3s_ToVec3f(&point, &points[currentPoint]);
+    if (currentPoint == 0) {
+        diffX = points[1].x - points[0].x;
+        diffZ = points[1].z - points[0].z;
+    } else if (currentPoint == pathCount - 1) {
+        diffX = points[pathCount - 1].x - points[pathCount - 2].x;
+        diffZ = points[pathCount - 1].z - points[pathCount - 2].z;
     } else {
-        phi_f12 = sp5C[idx + 1].x - sp5C[idx - 1].x;
-        phi_f14 = sp5C[idx + 1].z - sp5C[idx - 1].z;
+        diffX = points[currentPoint + 1].x - points[currentPoint - 1].x;
+        diffZ = points[currentPoint + 1].z - points[currentPoint - 1].z;
     }
 
-    func_8017B7F8(&sp30, RADF_TO_BINANG(func_80086B30(phi_f12, phi_f14)), &sp44, &sp40, &sp3C);
-    if (((this->actor.world.pos.x * sp44) + (sp40 * this->actor.world.pos.z) + sp3C) > 0.0f) {
-        sp50 = true;
+    func_8017B7F8(&point, RADF_TO_BINANG(func_80086B30(diffX, diffZ)), &px, &pz, &d);
+
+    if (((this->actor.world.pos.x * px) + (pz * this->actor.world.pos.z) + d) > 0.0f) {
+        reached = true;
     }
 
-    return sp50;
+    return reached;
 }
 
-s16 func_809895B4(Path* path, s32 idx, Vec3f* pos, f32* distSQ) {
+/**
+ * Returns the Y-rotation the dog should have to move to the next point along its path.
+ */
+s16 EnDg_GetYRotation(Path* path, s32 idx, Vec3f* pos, f32* distSQ) {
```

Comment:
```suggestion
s16 EnDg_GetYRotation(Path* path, s32 index, Vec3f* pos, f32* distSq) {
```

Reply:
Pretty sure I copied this from somewhere else, so it's going to be wrong in other places too. But let's fix this one for now to set a good example.